relay-server/src/services/projects/cache/project.rs

```diff
@@ -77,12 +77,17 @@ impl<'a> Project<'a> {
         let current_limits = self.rate_limits().current_limits();
 
         let quotas = state.as_deref().map(|s| s.get_quotas()).unwrap_or(&[]);
-        let envelope_limiter = EnvelopeLimiter::new(CheckLimits::NonIndexed, |item_scoping, _| {
-            Ok(current_limits.check_with_quotas(quotas, item_scoping))
-        });
+        let current_limits_clone = current_limits.clone();
+        let envelope_limiter =
+            EnvelopeLimiter::new(CheckLimits::NonIndexed, move |item_scoping, _| {
+                let current_limits_clone = current_limits_clone.clone();
 
-        let (mut enforcement, mut rate_limits) =
-            envelope_limiter.compute(envelope.envelope_mut(), &scoping)?;
+                async move { Ok(current_limits_clone.check_with_quotas(quotas, item_scoping)) }
+            });
+
+        let (mut enforcement, mut rate_limits) = envelope_limiter
+            .compute(envelope.envelope_mut(), &scoping)
+            .await?;
```

Comment:
This should not need to be a future. And we should also not even have it in the signature even if it technically never goes to redis to avoid it even happening in the future.

Reply:
I did try to make this work for both sync and async, but it’s not feasible. Besides duplicating the code, I see no better solution.

One approach I explored was to make the constructor accept a normal closure and convert it inside to return a `Future`, but since closures are anonymous, we can’t name the return value of the method.